presto-main/src/main/java/io/prestosql/cost/LocalCostEstimate.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.cost;
+
+import io.prestosql.sql.planner.plan.PlanNode;
+
+import java.util.stream.Stream;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.Double.NaN;
+
+/**
+ * Represents inherent cost of some plan node, not including cost of its sources.
+ */
+public class LocalCostEstimate
+{
+    private final double cpuCost;
+    private final double maxMemory;
+    private final double networkCost;
+
+    public static LocalCostEstimate unknown()
+    {
+        return of(NaN, NaN, NaN);
+    }
+
+    public static LocalCostEstimate zero()
+    {
+        return of(0, 0, 0);
+    }
+
+    public static LocalCostEstimate ofCpu(double cpuCost)
+    {
+        return of(cpuCost, 0, 0);
+    }
+
+    public static LocalCostEstimate ofNetwork(double networkCost)
+    {
+        return of(0, 0, networkCost);
+    }
+
+    public static LocalCostEstimate of(double cpuCost, double maxMemory, double networkCost)
+    {
+        return new LocalCostEstimate(cpuCost, maxMemory, networkCost);
+    }
+
+    private LocalCostEstimate(double cpuCost, double maxMemory, double networkCost)
+    {
+        this.cpuCost = cpuCost;
+        this.maxMemory = maxMemory;
+        this.networkCost = networkCost;
+    }
+
+    public double getCpuCost()
+    {
+        return cpuCost;
+    }
+
+    public double getMaxMemory()
+    {
+        return maxMemory;
+    }
+
+    public double getNetworkCost()
+    {
+        return networkCost;
+    }
+
+    /**
```

Comment:
it could do today, but won't be able to do this reasonably in the future.
So asking costcomparator to compare local costs, would be just hiding the fact we're hacking here. This hack is good, but i want to keep it explicit.

Reply:
how to remove the hack then?